javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java

```diff
@@ -57,6 +66,106 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
+                        exampleValue.put(serializedParameterName,
+                                ModelExampleUtil.getParameterExampleValue(
+                                        proxyMethodExample, serializedParameterName, parameter.getRequestParameterLocation()));
+                    }
+                }
+            }
+            IType type = methodParameter.getClientMethodParameter().getClientType();
+            IType wireType = methodParameter.getClientMethodParameter().getWireType();
+            return ModelExampleUtil.parseNode(type, wireType, exampleValue);
+        } else if (isFlattenParameter(convenienceMethod, methodParameter)) {
+            // flatten, no grouping
+            String realParameterName = convenienceMethod.getMethodTransformationDetails().iterator().next().getOutParameter().getName();
+            Map<String, Object> realParameterValue = (Map<String, Object>) proxyMethodExample.getParameters().get(realParameterName).getObjectValue();
+
+            IType type = methodParameter.getClientMethodParameter().getClientType();
+            IType wireType = methodParameter.getClientMethodParameter().getWireType();
+
+            Object flatteningParameterValue = null;
+            if (realParameterValue != null) {
+                flatteningParameterValue = realParameterValue
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> Objects.equals(
+                                CodeNamer.getEscapedReservedClientMethodParameterName(entry.getKey()),
+                                methodParameter.getClientMethodParameter().getName()))
+                        .findFirst()
+                        .map(Map.Entry::getValue).orElse(null);
+            }
+            return ModelExampleUtil.parseNode(type, wireType, flatteningParameterValue);
+        } else {
+            return ModelExampleUtil.parseNodeFromParameter(proxyMethodExample, methodParameter);
+        }
+    }
+
+    private boolean isGroupingParameter(ClientMethod convenienceMethod, MethodParameter methodParameter) {
+        List<MethodTransformationDetail> details = convenienceMethod.getMethodTransformationDetails();
+        if (CoreUtils.isNullOrEmpty(details) || details.size() <= 1) {
+            return false;
+        }
+
+        return details.stream().allMatch(
+                detail ->
+                        !CoreUtils.isNullOrEmpty(detail.getParameterMappings())
+                                && detail.getOutParameter() != null
+                                &&
+                                // same name
+                                Objects.equals(
+                                        CodeNamer.getEscapedReservedClientMethodParameterName(
+                                                detail.getParameterMappings().iterator().next().getInputParameter().getName()),
+                                        methodParameter.getClientMethodParameter().getName())
```

Comment:
again, I am not sure about this.

Reply:
Changed to check inputParameter's name against the client method parameter's name.